src/classes/ChildCreatorController.cls

```diff
@@ -4,17 +4,8 @@ public with sharing class ChildCreatorController {
 	public static SObject[] saveItems(SObject[] itemToBeSavedList, SObject[] itemToBeRemovedList){
 		system.debug('@@itemToBeSavedList '	+itemToBeSavedList);
 		system.debug('@@itemToBeRemovedList '	+itemToBeRemovedList);
-		List<SObject> toInsertList = new List<SObject>();
-		List<SObject> toUpdateList = new List<SObject>();
-		for(SObject item: itemToBeSavedList){
-			if(item.Id == null){
-				toInsertList.add(item);
-			} else {
-				toUpdateList.add(item);
-			}
-		}
-		insert toInsertList;
-		update toUpdateList;
+		
+		upsert itemToBeSavedList;
```

Comment:
That's strange, the unit test still works after I made the change. What exception do you get?